mentat/code_context.py

```diff
@@ -42,6 +43,7 @@ class CodeContextSettings:
     diff: Optional[str] = None
```

Comment:
(Referring to the _count_tokens_in_features function above this) I know you didn't add this in this PR, but why do we need a semaphore? I didn't think tiktoken made any http requests or used any outside resources at all

Reply:
I think it's about mentat opening too many files at once. I believe @biobootloader had an issue with that before the semaphore was added

Reply:
Ah, I see, that's unfortunate